From 8845035bb98d9676e23d0b6b754ec859dbe0d175 Mon Sep 17 00:00:00 2001
From: Thomas Merrell-ATM098 <thomasmerrell@motorola.com>
Date: Wed, 25 May 2016 12:30:43 -0500
Subject: [PATCH 906/959] IKFPS-1113: input: fps: Add timed wake lock to fps
 IRQ handler

This is to provide enough time or FPS HAL to service the event
and request the driver enable clocks and take control of the
wake lock.

Signed-off-by: Thomas Merrell-ATM098 <thomasmerrell@motorola.com>
Change-Id: I45afa46058920215aef261f7e339068ad6cc493b
Reviewed-on: http://gerrit.mot.com/866583
SLTApproved: Slta Waiver <sltawvr@motorola.com>
SME-Granted: SME Approvals Granted
Reviewed-by: Christopher Fries <cfries@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 drivers/input/misc/fpc1020_tee.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/input/misc/fpc1020_tee.c b/drivers/input/misc/fpc1020_tee.c
index f8f0600669a..82fb8090683 100644
--- a/drivers/input/misc/fpc1020_tee.c
+++ b/drivers/input/misc/fpc1020_tee.c
@@ -343,6 +343,7 @@ static irqreturn_t fpc1020_irq_handler(int irq, void *handle)
 {
 	struct fpc1020_data *fpc1020 = handle;
 
+	wake_lock_timeout(&fpc1020->wlock, msecs_to_jiffies(1000));
 	dev_dbg(fpc1020->dev, "%s\n", __func__);
 	sysfs_notify(&fpc1020->dev->kobj, NULL, dev_attr_irq.attr.name);
 	return IRQ_HANDLED;
@@ -433,6 +434,7 @@ static int fpc1020_probe(struct spi_device *spi)
 		goto exit;
 	}
 
+	wake_lock_init(&fpc1020->wlock, WAKE_LOCK_SUSPEND, "fpc1020");
 
 	fpc1020->clocks_enabled = false;
 	fpc1020->clocks_suspended = false;
@@ -451,7 +453,6 @@ static int fpc1020_probe(struct spi_device *spi)
 	/* Request that the interrupt should be wakeable */
 	enable_irq_wake(gpio_to_irq(fpc1020->irq_gpio));
 
-	wake_lock_init(&fpc1020->wlock, WAKE_LOCK_SUSPEND, "fpc1020");
 
 	rc = sysfs_create_group(&dev->kobj, &attribute_group);
 	if (rc) {
-- 
2.11.0

